libs/ktem/ktem/tags/pipelines.py

```diff
@@ -0,0 +1,245 @@
+from collections import defaultdict
+from pathlib import Path
+from typing import Generator
+
+from ktem.db.base_models import BaseTag, TagScope
+from ktem.db.models import engine
+from ktem.index.file.base import BaseFileIndexIndexing
+from ktem.index.file.pipelines import IndexDocumentPipeline, IndexPipeline
+from ktem.llms.manager import llms
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+from kotaemon.base import Document, DocumentWithEmbedding
+from kotaemon.llms import BaseLLM
+
+from ..db.base_models import TagProcessStatus
+from .crud import ChunkTagIndexCRUD, TagCRUD
+
+
+class MetaIndexPipeline(IndexDocumentPipeline):
+    """Meta index pipeline for tagging documents.
+    We reuse the logic from IndexDocumentPipeline and add custom
+    tagging procedure with .stream() method.
+    """
+
+    llm: BaseLLM
+    tags: list[str]
+
+    @property
+    def chunk_tag_index_crud(self) -> ChunkTagIndexCRUD:
+        return ChunkTagIndexCRUD(engine)
+
+    @property
+    def tag_crud(self) -> TagCRUD:
+        return TagCRUD(engine)
+
+    @classmethod
+    def resolve_tag_names(cls, tag_str: str) -> list[str]:
+        tags = []
+        if tag_str:

```

Comment:
```suggestion
        if tag_str == '':

```
since the `tag_str` is `str`, we should explicitly check if it's an empty string